### DISCUSSION TITLE
How do I delete a cache entry?

### DISCUSSION THREAD

**Discussion by vikaskedia** [CodeDredd/pinia-orm#711] [Q&A] [answered]
I am using cache as per the docs at: https://pinia-orm.codedredd.de/guide/repository/retrieving-data#caching

Suppose I get a list of users with a query like:

const users = useRepo(User).useCache().get()

After 1 minute the user changes his email address and I want to delete the cache entry.

How do I do that?

**Comment by CodeDredd** [comment, marked as answer]
Well there a three options.
1. You could just clear the complete cache by using `useRepo(User).cache().clear()`
2. If you know the cache key you can just delete the cached entry: `useRepo(User).cache().delete('key')`
3. Iterate through the cache entries and update the specific models. `useRepo(User).cache().forEach((key, value) => ...)`
